src/Symfony/Component/Cache/Adapter/PhpArrayAdapter.php

```diff
@@ -0,0 +1,367 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Adapter building static PHP files that will be cached by OPCache.
+ * This Adapter is in read-only if you use AdapterInterface methods.
+ * You can use the method "warmUp" to build the cache file.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class PhpArrayAdapter implements AdapterInterface
+{
+    private $file;
+
+    /**
+     * @var array
+     */
+    private $values;
+
+    /**
+     * @var \Closure
+     */
+    private $createCacheItem;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $fallbackPool;
+
+    /**
+     * @param string $file The PHP file were values are cached.
+     *
+     * @return CacheItemPoolInterface
+     */
+    public static function createOnPhp7($file, CacheItemPoolInterface $fallbackPool)
```

Comment:
I don't like the method name. I suppose that this will also work on PHP 8, right? So including a version in a method name sounds fragile to me. But then, I don't understand the usage, so I might be wrong.